pJanks/overlook
# Building customer class and tests.

### What's this PR do?
This adds catches to my fetch calls. It also builds out all or most of the tests I need for the Customer class and passes them. I am testing all default properties and methods. I am currently having trouble trying to figure out how to check the date property though.
### Where should the reviewer start?
I really want feedback on my Customer-tests with emphasis on if they are robust enough and if they thoroughly test method paths. I would also like feedback on things that may be refactored in the Customer class
### How should this be manually tested?
Review the files changed and pull down the branch to make sure tests pass and there aren't any bugs.
### Potential changes?
My next step is to build out a child class of Customer for Manager and write the tests to get Manger's exclusive tests and properties tested.

@khalidwilliams
@hannahhch
